Makefile

```diff
@@ -46,23 +46,23 @@ clean-conda-env:
 	rm -f build/conda-env.${CONDA_ENV_NAME}.build-stamp
 
 .PHONY: check
-check: pycodestyle pydocstyle pylint mypy # cspell licenseheaders markdown-link-check
+check: black pydocstyle pylint mypy # cspell licenseheaders markdown-link-check
 
-.PHONY: pycodestyle
-pycodestyle: conda-env
-pycodestyle: build/pycodestyle.mlos_core.${CONDA_ENV_NAME}.build-stamp
-pycodestyle: build/pycodestyle.mlos_bench.${CONDA_ENV_NAME}.build-stamp
-pycodestyle: build/pycodestyle.mlos_viz.${CONDA_ENV_NAME}.build-stamp
+.PHONY: black
+black: conda-env
+black: build/black.mlos_core.${CONDA_ENV_NAME}.build-stamp
+black: build/black.mlos_bench.${CONDA_ENV_NAME}.build-stamp
+black: build/black.mlos_viz.${CONDA_ENV_NAME}.build-stamp
 
 
-build/pycodestyle.mlos_core.${CONDA_ENV_NAME}.build-stamp: $(MLOS_CORE_PYTHON_FILES)
-build/pycodestyle.mlos_bench.${CONDA_ENV_NAME}.build-stamp: $(MLOS_BENCH_PYTHON_FILES)
-build/pycodestyle.mlos_viz.${CONDA_ENV_NAME}.build-stamp: $(MLOS_VIZ_PYTHON_FILES)
+build/black.mlos_core.${CONDA_ENV_NAME}.build-stamp: $(MLOS_CORE_PYTHON_FILES)
+build/black.mlos_bench.${CONDA_ENV_NAME}.build-stamp: $(MLOS_BENCH_PYTHON_FILES)
+build/black.mlos_viz.${CONDA_ENV_NAME}.build-stamp: $(MLOS_VIZ_PYTHON_FILES)
 
-build/pycodestyle.%.${CONDA_ENV_NAME}.build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp setup.cfg
-	# Check for decent pep8 code style with pycodestyle.
-	# Note: if this fails, try using autopep8 to fix it.
-	conda run -n ${CONDA_ENV_NAME} pycodestyle $(filter-out setup.cfg,$+)
+build/black.%.${CONDA_ENV_NAME}.build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp setup.cfg
+	# Check for decent pep8 code style with black.
+	# Note: if this fails, try using black to fix it.
+	conda run -n ${CONDA_ENV_NAME} black --check $(filter-out setup.cfg,$+)
```

Comment:
Propose two different rules:
1. to check only (for use in CI)
2. one to run black on modified *.py files locally
In either case please restrict it to *.py files only.
I can help write the Makefile rules if you need.

Reply:
Additionally, for parallelization, we can run `black` against individual projects.

Check the top of the `Makefile` for some rules on `*.py` file discovery.

Reply:
Black does not format non-py files, these changes have been reverted. Added make black rule